[DOCS-15059] Add Agent-side Tag Aggregation guide (Preview) (#38245)
DataDog/documentation: content/en/metrics/guide/agent-side-tag-aggregation.md

  tag: "Documentation"
  text: "Agent-Side Filtering for Custom Metrics"
- link: "/metrics/guide/tag-indexing-rules/"
  tag: "Documentation"
  text: "Tag Indexing Rules"
- link: "/metrics/metrics-without-limits/"
  tag: "Documentation"
  text: "Metrics without Limits™"
- link: "/remote_configuration/"
  tag: "Documentation"
  text: "Remote Configuration"
---
{{< callout url="#" header="false" btn_hidden="true" >}}
Agent-side tag aggregation is in Preview.
{{< /callout >}}

## Overview

Agent-side tag aggregation lets you reduce custom metric volume. It filters out unwanted tags and re-aggregates selected custom metrics in the Datadog Agent before those metrics are sent to Datadog.

## Prerequisites

Creating and updating Agent-side tag aggregation rules requires the [`metric_tags_write`][3] RBAC permission. All users can view existing tag aggregation rules.

Before you create Agent-side tag aggregation rules, complete the steps described in the following sections:

1. [Enable Remote Configuration](#enable-remote-configuration) on the API keys used by your Agents.
2. [Enable the Agent Data Plane](#enable-the-agent-data-plane).

> Agent-side tag aggregation is supported on Agent version 7.80.0 and later. Datadog recommends version 7.80.2 or later, which includes additional troubleshooting information in Agent flares.

Reviewer: We now recommend 7.81.0 or later

```suggestion
Agent-side tag aggregation is supported on Agent version 7.80.0 and later. Datadog recommends version 7.81.0 or later, which includes broad backward compatibility of the Agent Data Plane with pre-existing Agent configuration options.
```

### Enable Remote Configuration

Remote Configuration must be enabled for your organization and for the API keys used by your Agents. For more information about the Remote Configuration security model, see the [Remote Configuration security whitepaper][4].

Comprehensive instructions are available in the [Remote Configuration documentation][5]. The steps are summarized below:

1. With the [`org_management`][6] permission, enable Remote Configuration for your organization.
2. With the [`api_keys_write`][7] permission, enable the Remote Configuration capability on the API keys used by your Agents.
3. Restart your Agents after enabling Remote Configuration on the API keys. The API key capability change takes effect only after the Agents restart.

### Enable the Agent Data Plane

> Upgrade your Agents to version 7.80.0 or later (7.80.2 or later recommended) and enable the Agent Data Plane. Agent-side tag aggregation requires the Agent Data Plane, a high-performance data pipeline that lowers the combined resource usage of Agent processes and containers.

Reviewer: (recommend 7.81.0+)

```suggestion
Upgrade your Agents to version 7.80.0 or later (7.81.0 or later recommended) and enable the Agent Data Plane. Agent-side tag aggregation requires the Agent Data Plane, a high-performance data pipeline that lowers the combined resource usage of Agent processes and containers.
```

When the Agent Data Plane is enabled:

- In Kubernetes environments, a new `agent-data-plane` container runs.
- In Linux environments, a new `agent-data-plane` systemd service and process runs.

{{< tabs >}}
{{% tab "Datadog Operator" %}}

With Operator version 1.27.0 or later, add the following to your `DatadogAgent` CRD:

{{< code-block lang="yaml" >}}
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  features:
    dataPlane:
      enabled: true # enable the Agent Data Plane feature
  override:
    nodeAgent:
      image:

>         tag: "7.80.2" # use 7.80.2 or later

Reviewer: ```suggestion
        tag: "7.81.0" # use 7.81.0 or later
```

{{< /code-block >}}

{{% /tab %}}
{{% tab "Helm" %}}

With Helm chart version 3.202.5 or later, add the following to your `values.yaml`:

{{< code-block lang="yaml" >}}
datadog:
  dataPlane:
    enabled: true
  dogstatsd:
    enabled: true # enable the Agent Data Plane feature
agents:
  image:

>     tag: "7.80.2" # use 7.80.2 or later

Reviewer: ```suggestion
    tag: "7.81.0" # use 7.81.0 or later
```

{{< /code-block >}}

{{% /tab %}}
{{% tab "Linux VM" %}}

> With Agent version 7.80.0 or later (7.80.2 or later recommended), add the following to your `datadog.yaml`:

Reviewer: ```suggestion
With Agent version 7.80.0 or later (7.81.0 or later recommended), add the following to your `datadog.yaml`:
```

## Create Agent-side tag aggregation rules

After Remote Configuration and the Agent Data Plane are enabled, you can configure Agent-side tag aggregation from the Datadog UI.

To create a rule:

1. Navigate to the [Metric Summary][8] page or the [Custom Metrics Volume][9] page.
2. Select one or more custom metrics for Agent-side tag aggregation.

Your tag aggregation rules are visible from the [Metric Settings][10] page, where you can also create and edit them directly.

## Preview limitations

The initial Preview release has the following limitations:

- Up to 500 metric names can be configured for Agent-side tag aggregation.
- Only DogStatsD distribution metrics and DogStatsD count and rate metrics are supported.
- Windows and macOS are not supported.
- Managing Agent-side tag aggregation rules through the API is not supported.

When Agent-side tag aggregation rules are configured, Agent Data Plane resource usage may increase by up to 10% CPU and up to 25% resident set size (RSS). These limits are intended to minimize the resource usage impact on the Agent during the Preview.

> ## Example: query impact on count and rate metrics

Reviewer 2: ```suggestion
## Example: Query impact on count and rate metrics
```

Tiny style nit here per https://datadoghq.atlassian.net/wiki/spaces/WRITING/pages/5341577434/Capitalization#Capitalization-for-titles-and-headings
